clang/test/SemaSYCL/sycl-device-intel-max-global-work-dim-template.cpp

```
@@ -61,11 +61,30 @@ template <int N>
 // expected-error@+1{{'max_global_work_dim' attribute requires a non-negative integral compile time constant expression}}
 [[intel::max_global_work_dim(N)]] void func3() {}
 
+// Test that checks template instantiations for different argument values.
+template <int size>
+[[intel::max_global_work_dim(1)]] void func4(); // expected-note {{previous attribute is here}}
+
+template <int size>
+[[intel::max_global_work_dim(size)]] void func4() {} // expected-warning {{attribute 'max_global_work_dim' is already applied with different arguments}}
+
+// Test that checks template instantiations for same argument values. We don't duplicate the attribute with the same arguments.
+template <int size>
+[[intel::max_global_work_dim(2)]] void func5();
+
+template <int size>
+[[intel::max_global_work_dim(size)]] void func5() {}
+
+
```

Add test for no arguments mismatch. Silently ignore duplicate attribute here.